src/plugins/data/public/search/session/session_service.ts

```diff
@@ -241,24 +245,34 @@ export class SessionService {
       );
       return;
     }
-
+    this.cleanupPreviousSession();
     this.state.transitions.clear();
     this.searchSessionInfoProvider = undefined;
     this.searchSessionIndicatorUiConfig = undefined;
   }
 
+  private async cleanupPreviousSession() {
+    const { pendingSearches, sessionId, isRestore, isStored } = this.state.get();
+    if (sessionId && !isRestore && !isStored) {
+      pendingSearches.forEach((s) => {
+        s.abort();
+      });
+      await this.sessionsClient.cancel(sessionId).catch(() => {});
+    }
+  }
+
   /**
    * Request a cancellation of on-going search requests within current session
    */
   public async cancel(): Promise<void> {
-    const isStoredSession = this.state.get().isStored;
-    this.state.get().pendingSearches.forEach((s) => {
+    const { pendingSearches, sessionId } = this.state.get();
+    if (!sessionId) return;
+
+    pendingSearches.forEach((s) => {
       s.abort();
     });
+    await this.sessionsClient.cancel(sessionId).catch(() => {});
     this.state.transitions.cancel();
-    if (isStoredSession) {
```

Comment:
Replaced deletion with cancellation.
The cleanup task will take care of deletion https://github.com/elastic/kibana/pull/99967